### PR TITLE
feat: key details

### PR DESCRIPTION
**Pull request by guru-web3**

### CODE REVIEW

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (metalurgical) [inline Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift]: Not sure that we want to mix details of the metadata key with the signing key, these are different things?

Consider separate methods. Also, reconstructing the key would return KeyDetails by default, so why the need to fetch it again?

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (metalurgical) [inline Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift:154]: This should be checked first.

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (metalurgical) [inline Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift:159]: You'll also need the tss index. 

Consider returning this as a sub struct of:
```
tss_tag
tss_index
verifier
verifier_id
```

However what you return here will depend on what you want to use it for, is it just for display?

Comment (metalurgical) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift:159]: Since multiple tss-tag can exist, this struct will also not be singular but a sub item in an array.

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (guru-web3) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift]: in mpc web we have the same logic to get tss pub and attach
for ref
`  
  public getKeyDetails(): MPCKeyDetails {
    this.checkReady();
    const tkeyDetails = this.tKey.getKeyDetails();
    const tssPubKey = this.state.tssPubKey ? Point.fromBufferSEC1(this.state.tssPubKey).toTkeyPoint() : undefined;

    const factors = this.tKey.metadata.factorPubs ? this.tKey.metadata.factorPubs[this.tKey.tssTag] : [];
    const keyDetails: MPCKeyDetails = {
      // use tkey's for now
      requiredFactors: tkeyDetails.requiredShares,
      threshold: tkeyDetails.threshold,
      totalFactors: factors.length + 1,
      shareDescriptions: this.tKey.getMetadata().getShareDescription(),
      metadataPubKey: tkeyDetails.pubKey,
      tssPubKey,
    };
      return keyDetails;
    }
  `

Comment (metalurgical) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift]: Yes I get that. It doesn't answer the original questions though. This may also need to be changed there depending on the answers here.

Comment (ieow) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift]: Let get it match with the web version first. (v2)
we can file pr and discuss if the changes is needed for mpc core kit web ( especially v3 that is in working)

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (guru-web3) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift:154]: done

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (guru-web3) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift:159]: following the same return type as web version let me know @ieow we can implement this too in web first?

Comment (ieow) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift:159]: I think we would not encourage client to use the tags as we now have soft-nonce index for the multi-wallet.

File: Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift
Comment (metalurgical) [reply Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift]: @ieow Ok, approve it if you're happy with it. Think we'll have to come back it change it though, however not blocking this PR from going in.

**Review by ieow — APPROVED** [review]
lgtm

**Pull request merged**
